lib/broken_record/slack_notifier.rb

```diff
@@ -1,27 +1,52 @@
 require 'net/http'
 module BrokenRecord
   class SlackNotifier
-    def initialize(options)
-      @options = BrokenRecord::Config.slack_options.merge(options)
+    def initialize(initial_options)
+      @options = default_options
+      @options = @options.merge(initial_options)
+      @options = @options.merge(BrokenRecord::Config.slack_options)
     end
 
     def send!(message)
+      return unless @options[:summary]
       if defined?(Rails) && Rails.env.development?
         puts message
       else
-        slack_params = options.merge(text: message)
+        slack_params = @options.merge(text: message)
         uri = URI('https://slack.com/api/chat.postMessage')
         uri.query = URI.encode_www_form(slack_params)
         Net::HTTP.get(uri)
       end
     end
 
-    private
+    def send_snippet!(snippet_body, snippet_title)
+      return unless @options[:snippet]
+      if defined?(Rails) && Rails.env.development?
+        puts message
+      else
+        slack_params = @options.merge({
+          title: snippet_title,
+          initial_comment: ':heavy_exclamation_mark: @benefitsvm: The validation job failed!',
+          icon_emoji: ':x:',
+          channels: @options[:channel]
+        })
 
-    def options
-      default_options.merge(@options)
+        # Upload file to slack via multipart/form-data
+        snippet_file = Tempfile.new
+        begin
+          snippet_file.write(snippet_body)
+          snippet_file.rewind
+          encoded_params = slack_params.map{ |key, value| "-F \"#{key}=#{value}\"" }.join(' ')
+          `curl -F file=@#{snippet_file.path} #{encoded_params} https://slack.com/api/files.upload`
```

Comment:
Assuming that you tested this, as I don't know anything about making curl requests like this :)